src/components/BaseComponent.js

```diff
@@ -105,14 +110,31 @@ export default class BaseComponent extends Component {
     /* IMPLEMENT */
   }
 
+  /**
+   * If there is an adaptor on file, adapt data
+   */
+  adaptData(data) {
+    let _data = Object.assign([], data);
+    if (this.state.lib && this.state.funcType) {
+      const adaptor = new Adaptor();
```

Comment:
This will create a new adaptor each time adaptData it's called. We should move this to the BaseComponent constructor and assign that to a instance variable.